content/blog/2023/09/25/2023-09-25-building-jenkinsio-with-alternative-tools.adoc

```diff
@@ -0,0 +1,72 @@
+. <<Acknowledgments>>
+. <<Useful Links>>
+
+== About Project
+
+Jenkins.io is generated as a static website using Awestruct from AsciiDoc sources. One of the drawbacks of the current build method is that the technical documentation is not Versioned. It is thus not possible to view the documentation for a given Jenkins version. Only the latest can be viewed. This can lead to unnecessary confusion and is a worse experience than many other documentation sites like the git site, FreeBSD, and others.. Other drawback of Awestruct is that it is greatly under maintained since last two years and people from jenkins-infra are currently maintaining it. Relying on awestruct is not a good thing for a longer run. 
+
+The preferred set of tools that will replace Awestruct are https://antora.org/[Antora] for it's out of the box versioning functionality and https://www.gatsbyjs.com/[Gatsby] for it's super fast builds.
+
+The migration was performed in the following 2 coding phases:
+
+== Phase 1
+
+In this phase, the following tasks were completed:
+
+* I used Antora to migrate the User documentation which was versioned and Developer documentation which was non-versioned while keeping all the functionalities that we have on jenkins.io right now.
+* I migrated Tutorials, Guides, Solution Pages which were all non-versioned.
+* I fixed Interpage Linking across all the pages.
+
+
+You can watch me talk about the project in the https://youtu.be/W4eSVCTmqb8[Midterm Presentation here].
+
+== Phase 2
+
+In this phase, the following tasks were completed:
+
+* I used Strapi CMS with Gatsby. I would also have to integrate Git Workflow here which was not suggested by Mentors. Hence, It was not the perfect fit for Jenkins.
+* I then moved to Gatsby and Developed the Blogs Feature.
+* I developed other pages like LTS Changelog, LTS Weekly, Download page and many more.
+
+Here's the link to https://youtu.be/W4eSVCTmqb8[Midterm Presentation] where you can watch me talk about the project's second half.
+
+== Future Improvements
+
+- Some pages like Changelogs, Pipeline step reference in documentation still need more refining and are missing some features that exist on the pages on Jenkins.io. I will be keep working on the project to refine them and make it ready as a whole to be called a drop-in replacement to Jenkins.io.
+
+== Acknowledgments
+
+I'm grateful that I got the opportunity to contribute to Jenkins under Google Summer of Code. I can say with confidence that Jenkins taught me a lot and I love learning so I don't plan to stop contributing to Jenkins any soon. I was able to pull off the project to this extent only because of my mentors,link:/blog/authors/krisstern/[Kris Stern], link:/blog/authors/MarkEWaite/[Mark Waite], link:/blog/authors/iamrajiv/[Rajiv Ranjan Singh], link:/blog/authors/yiminggong/[Yiming Gong] constant support and guidance throughout the project. Because of their valuable feedback and insights into the project helped me a lot.
```

Comment:
Use the author macro rather than making a link to the author.  Let the existing macros do more of the work.
```suggestion
I'm grateful that I got the opportunity to contribute to Jenkins under Google Summer of Code.
I can say with confidence that Jenkins taught me a lot and I love learning so I don't plan to stop contributing to Jenkins any soon.
I was able to pull off the project to this extent only because of my mentors, author:krisstern[Kris Stern], author:MarkEWaite[Mark Waite], author:iamrajiv[Rajiv Ranjan Singh], and author:yiminggong/[Yiming Gong].
I am grateful for their constant support and guidance throughout the project.
Their valuable feedback and insights into the project helped me a lot.
```